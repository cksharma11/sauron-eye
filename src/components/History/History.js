import React from "react";
import Reports from "../Reports/Reports";

const getUsername = () => {
  const search = window.location.search;
  const urlParams = new URLSearchParams(search);
  return urlParams.get("username");
};

const History = ({ pushers }) => {
  const username = getUsername();
  const details = pushers[username];

  return <Reports details={details} username={username} />;
};

export default History;
