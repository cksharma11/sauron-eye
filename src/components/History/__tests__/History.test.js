import React from "react";
import { shallow, configure } from "enzyme";
import mockPushers from "../../../data/mock.json";
import Adapter from "enzyme-adapter-react-16";
import History from "../History";

configure({ adapter: new Adapter() });

describe("History", () => {
  it("should render properly", () => {
    const pushers = {
      sha: "1234HSA",
      total: 15,
      passed: 10,
      failed: 5,
      failedSuites: [
        "1 should be equal to 1",
        "2 should be equal to 2",
        "3 should be equal to 3",
        "4 should be equal to 4",
        "5 should be equal to 5"
      ],
      time: "sometime",
      project: "head",
      name: "testName"
    };
    
    const wrapper = shallow(<History pushers={pushers} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
