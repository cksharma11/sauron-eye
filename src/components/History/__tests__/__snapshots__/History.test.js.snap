// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History should render properly 1`] = `"<Reports details={[undefined]} username={{...}} />"`;
